### PR TITLE
[ParticleMechanicsApplication]/add option for time step table

### PR DESCRIPTION
**Pull request by VeronikaSinger**
This small PR adds the option for a time step table in MPM and fixes a small error in the vtk output

### CODE REVIEW

**Review by tteschemacher — DISMISSED** [review]
looks good. Maybe a test - just to easily understand how to use it.

File: applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py [hunk 1]
```diff
@@ -435,7 +435,17 @@ def _SetBufferSize(self):
...
     def __ComputeDeltaTime(self):
-        return self.settings["time_stepping"]["time_step"].GetDouble()
+        if self.settings["time_stepping"].Has("time_step"):
+            return self.settings["time_stepping"]["time_step"].GetDouble()
+        elif self.settings["time_stepping"].Has("time_step_table"):
+            current_time = self.grid_model_part.ProcessInfo[KratosMultiphysics.TIME]
+            time_step_table = self.settings["time_stepping"]["time_step_table"].GetMatrix()
+            tb = KratosMultiphysics.PiecewiseLinearTable()
+            for interval in range(time_step_table.Size1()):
+                tb.AddRow(time_step_table[interval, 0], time_step_table[interval, 1])
```
Comment (philbucher) [inline applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]:
```suggestion
            time_step_table = self.settings["time_stepping"]["time_step_table"].GetMatrix()
            tb = KratosMultiphysics.PiecewiseLinearTable(time_step_table)
```

Comment (philbucher) [reply applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]: this should work, can you try?

Comment (VeronikaSinger) [reply applications/ParticleMechanicsApplication/python_scripts/mpm_solver.py]: cool! Thanks that works

**Comment by VeronikaSinger** [comment]
I'll also add a small example

**Review by philbucher — APPROVED** [review]
👍

**Comment by tteschemacher** [comment]
thanks a lot for the test!

**Review by tteschemacher — APPROVED** [review]

**Pull request merged**